### ISSUE TITLE
Crash on launch

### ISSUE THREAD

**Issue by jcgavula** [JohnCoates/Aerial#1179]
#### Required information
In order to help us sort your issue, we ask that you provide the following information:
- [ ] Mac model: iMac (27-inch, Late 2013)
- [ ] macOS version: macOS 10.15.7
- [ ] Monitor setup: Built-in 27"

If appropriate, please enable `Debug mode` and `Log to disk` in `Advanced` tab and replicate your bug, then attach the `AerialLog.txt` file (You can access this file through the Advanced tab).

#### Description of issue / Feature request

When I launch Aerial Companion it shows up briefly in Activity Monitor and then disappears. Looking in Console I see the output in the attached file.

TL;DR: Aerial Companion crashes on launch.
[Aerial crash on launch.txt](https://github.com/JohnCoates/Aerial/files/6519637/Aerial.crash.on.launch.txt)

**Comment by glouel** [comment]
Hey @jcgavula 

Sorry for the late reply, been a long day. I'm wondering, by any chance are you using some sort of MDM management tool (like JAMF) on this mac ? 

I'm wondering what is causing the too many corpses thing. 

I'll investigate and let you know, but it may look like it's usually a filesystem issue, you may want to read this : https://apple.stackexchange.com/questions/322509/opendirectoryd-too-many-corpses-being-created

The other cause seems to be bad permissions on some folders set by JAMF and some other MDM tools. 

Let me know more about your setup and we'll try to narrow it down. Thanks !

**Comment by jcgavula** [comment]
No worries. I know everyone gets busy, and while screensavers are awesome, other things take precedence. This is my personal Mac, so no JAMF or other MDM tools.

I followed the steps in the link to rename mbr_cache, and after rebooting I'm getting a crash report when I launch Aerial Companion. I'm attaching in the hopes that it'll help.

[Aerial Companion_2021-05-21-170903_Johns-iMac.crash.zip](https://github.com/JohnCoates/Aerial/files/6525073/Aerial.Companion_2021-05-21-170903_Johns-iMac.crash.zip)

**Comment by jcgavula** [comment]
I have a feeling it's something in my user profile - created a new user, and it works just fine. Does the user have to be an admin account to run Aerial Companion?

**Comment by glouel** [comment]
I’m not 100% sure but it might, I’ll make a non admin account tomorrow and check what happens and let you know.

**Comment by jcgavula** [comment]
Just a little more input: in Terminal as root I copied the Aerial.saver bundle from my admin account's ~/Library/Screen Savers/ directory over to /Library/Screen Savers/ so it's available globally. Now it works in my non-admin account. Weird.

**Comment by glouel** [comment]
Thanks ! I'll give a shot to a non privileged account today, but just to clarify a bit how it works, the issue here is purely with Aerial Companion which is separate (and what crashes for you) from the saver, so it's not surprising the saver itself works on your other account. The saver itself is its own bundle. 

Companion's role is a couple of thing, first one is install the saver (and you can either copy it to `~/Library/Screen Savers/` or to the global one, I default to each user's folder as the global one can be misbehaving on some subtle issues), second is keep it up to date. The whole purpose of Companion is to make it easier to auto update but one can manually get the latest release of the screensaver in zip form and install that. I probably need to reword a bit the download page to make that clear to everyone.

One issue with screen saver sandboxing is that each user's screensaver data is segregated and you can no longer have a common folder for all users where Aerial could download videos and share it with every user accounts. It's a bit of a mess and I hope they fix it next month 🙈

In any case, Aerial (the screensaver itself) should run fine on your other account in the meantime, but I'll still try to dig into the issue with Companion. Thanks !

**Comment by glouel** [comment]
So I put a new build up here (see my previous message/edit), I'm still not sure this will fix your issue but if you're willing to try that would be helpful. Thanks !

https://github.com/glouel/AerialCompanion/releases/download/v1.2.2/AerialCompanion.dmg

**Comment by jcgavula** [comment]
Thanks for the new build. Just tried the new 1.2.2 image - no dice. Console output and crash report attached.

[Aerial Companion_2021-05-23-080112_Johns-iMac.crash.zip](https://github.com/JohnCoates/Aerial/files/6528023/Aerial.Companion_2021-05-23-080112_Johns-iMac.crash.zip)
[Aerial 1.2.2_console_output.txt](https://github.com/JohnCoates/Aerial/files/6528024/Aerial.1.2.2_console_output.txt)

**Comment by glouel** [comment]
@jcgavula sorry it took me so long to reply to you, I had a look when you posted the logs but didn't see anything. I just had a second look today and found this which I believe may be the core of the issue : 

```
error	08:01:05.956933-0400	tccd	Prompting policy for hardened runtime; service: kTCCServiceAppleEvents requires entitlement com.apple.security.automation.apple-events but it is missing for ACC:{ID: com.glouel.AerialUpdater, PID[93292], auid: 501, euid: 501, binary path: '/Applications/Aerial Companion.app/Contents/MacOS/Aerial Companion'}, REQ:{ID: com.apple.appleeventsd, PID[413], auid: 55, euid: 55, binary path: '/System/Library/CoreServices/appleeventsd'}
```

Right now I have no idea about why that would kill Companion, but it might be around that. Companion is notarized though so I'm not 100% sure. The only thing I can think of is that maybe there's something that won't let you run unsandboxed app... That or, can you check in Preferences/security if you are not disallowing non app store apps by any chance ? See here : 

![Capture d’écran 2021-06-09 à 19 52 09](https://user-images.githubusercontent.com/37544189/121404614-65fa2a80-c95c-11eb-9a3a-9a4f8d158d2e.png)

**Comment by glouel** [comment]
I tried changing a couple of things but Aerial is already compiled with hardened runtime enabled. I tried to add some extra stuff just in case, if you have a sec, check message above and give that build a try : https://github.com/glouel/AerialCompanion/releases/download/v1.2test1/Aerial.Companion.app.zip

**Comment by jcgavula** [comment]
Thanks for the effort - no worries about the speed. It's not like you don't have other things to do. 

I checked my security settings: they're set to App Store and developers. 

I just tried v1.2test1, and am uploading the crash log and the console output.

In the meantime, I've dropped the screensaver bundle in /Library/Screen Savers, and Aerial is working like that, so I'm doing pretty good for now.

Thanks for all your work!
![Screen Shot 2021-06-10 at 8 45 40 AM](https://user-images.githubusercontent.com/84552354/121527596-91b2fa00-c9c8-11eb-9d04-0b6ead08ba9f.png)
[Aerial Companion_2021-06-10-084343_Johns-iMac.crash.zip](https://github.com/JohnCoates/Aerial/files/6631477/Aerial.Companion_2021-06-10-084343_Johns-iMac.crash.zip)
[Aerial console 2021_06_10.txt](https://github.com/JohnCoates/Aerial/files/6631483/Aerial.console.2021_06_10.txt)

**Comment by glouel** [comment]
Thanks for checking and following up. I think I was too focused on that console line, because I double checked here and I get the same exact warning (but no crash) : 

<img width="2174" alt="Capture d’écran 2021-06-10 à 15 13 15" src="https://user-images.githubusercontent.com/37544189/121531094-62b77b00-c9fe-11eb-852a-7e95743e5e14.png">

Interestingly there are three errors but macOS doesn't want to share them with me... Those are categorized as NIB loading issues, I'm not sure that's relevant. 

I'll still keep an eye open on this, but I think for the moment just using the bundles as you did will have to do :(

Thanks !

**Comment by imjeffro** [comment]
Hi,
I know this is an old thread but I seem to be having the same/similar issue. Just started with Aerial today.  Initial setup seemed to go ok, but then after playing with it for a bit I found that the options in the screensaver had gone blank. I attempted a reinstall, then to delete all the Aerial-related files and reinstall. I couldn't ever get the companion app to work again. So I found the download for the standalone screensaver app and installed that, which was successful. Now the current status is that the screensaver app is working fine but the Companion app crashes on launch.  
iMac Pro, OS 12.
[Aerial Companion-2022-01-21-144615 .txt](https://github.com/JohnCoates/Aerial/files/7916584/Aerial.Companion-2022-01-21-144615.txt)
1

**Comment by glouel** [comment]
Good to hear about the screensaver but it's puzzling about Companion.

if you can send me a crash log, and perhaps more importantly, eventual warning/errors that pop in console when you launch it, that may help me narrow it down !

**Comment by imjeffro** [comment]
attached the latest crash report from Console for aerial companion. if there's something else I can send, can you point me to instructions for doing it?  Thank you.
[Aerial Companion-2022-02-15-110905.ips.zip](https://github.com/JohnCoates/Aerial/files/8073296/Aerial.Companion-2022-02-15-110905.ips.zip)

**Comment by glouel** [comment]
@imjeffro thanks, I saw something new in that log I think that could be it. Can you tell me which locale you've set in macOS  (or which country you are from) ? I may have some issues with some localization stuff. Thanks !

**Comment by imjeffro** [comment]
Region is United States.
<img width="736" alt="Screen Shot 2022-02-15 at 11 45 40 AM" src="https://user-images.githubusercontent.com/8517940/154119078-3c659e60-09b5-454e-a8a6-fde845b57d50.png">
)

**Comment by imjeffro** [comment]
I had a hunch you were hoping for an exotic location to explain it ;)

~/Library/Application Support/AerialUpdater/ does not exist on my computer!

**Comment by imjeffro** [comment]
Bingo!  I created the folder, then launched Aerial Companion and it appeared in the taskbar. It updated the screensaver to 3.0.6, too.
It's possible that I deleted that folder during a manual uninstall/reinstall attempt and that's what caused the issue. I don't know why it didn't recreate the folder on its own after that.

Thank you! Sent you a coffee.

**Comment by glouel** [comment]
Yes ! The weird thing is, it should have created it again, so I'll try to have a look if maybe I messed up something.

And many thanks for the coffee, take care !

**Comment by jcgavula** [comment]
HOT DOG! Manually creating the Application Support directory fixed my issue, too! I'd been getting by manually updating the .saver file. Glad to have an answer for this.

**Comment by glouel** [comment]
Thanks for confirming @jcgavula ! Just to be clear, did you just create the Aerial directory in `Application Support`, or did you for some reason not have an `Application Support` at all ? I think maybe that could happen on a fresh install, that would definitely explain the bug (which I believe I fixed for next time).

Again thanks to you both for your patience on this, sorry it took me so long to figure out.

**Comment by jcgavula** [comment]
Updated my comment for clarity. And no worries on the timing - sometime 1's and 0's get fiddly. The world needs more time and patience, and I'm happy to share both.
